src/input/plugins/YtdlInputPlugin.cxx

```diff
@@ -0,0 +1,87 @@
+#include "config.h"
+#include "lib/ytdl/YtdlParser.hxx"
+#include "YtdlInputPlugin.hxx"
+#include "CurlInputPlugin.hxx"
+#include "../AsyncInputStream.hxx"
+#include "../ProxyInputStream.hxx"
+#include "../InputPlugin.hxx"
+#include "config/ConfigGlobal.hxx"
+#include "config/Block.hxx"
+#include "tag/TagBuilder.hxx"
+#include "tag/Tag.hxx"
+#include "event/Call.hxx"
+#include "event/Loop.hxx"
+#include "thread/Cond.hxx"
+#include "util/Alloc.hxx"
+#include "util/Domain.hxx"
+#include "Log.hxx"
+
+#include <cinttypes>
+
+#include <assert.h>
+#include <string.h>
+
+static constexpr Domain ytdl_domain("youtube-dl");
+
+static void
+input_ytdl_init(const ConfigBlock &block)
+{
+	// TODO: a configuration should allow overriding the curl input plugin
+	// this plugin would then accept http/https URIs as well
+}
+
+static void
+input_ytdl_finish() noexcept
+{
+}
+
+struct YtdlUriFlags {
+	bool notags;
+
+	YtdlUriFlags() : notags(false) { }
+};
+
+static std::string ytdl_uri(const char *uri, YtdlUriFlags* flags) {
```

Comment:
Bad C++: must use reference instead of pointer for obligatory pointer parameter; and since this is purely a return value, it should be declared as a return value and not as a parameter.